### PR TITLE
feat: Allow dangerous html based on config variable

### PR DESCRIPTION
**Pull request by MrwanBaghdad**
### Summary of Changes

Added configuration variable to allow rendering of dangerous HTML in the programmatic description for table detail pages

### Documentation

The use case is that we'd like to enable embedding in the programmatic descriptions. As the programmatic descriptions are populated by an ingestion pipeline that can validate its safety and is not editable by any web user; a requirement missing in the table description were editing may be allowed and it won't be safe to enable such functionality of rendering dangerous HTML

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by dorianj** [comment]
Do we even need to use `EditableText` for this, given that editable is false? Can we include `ReactMarkdown` directly? I think that would assuage the concern of turning `EditableText` into a footgun?

**Comment by Golodhros** [comment]
> Do we even need to use `EditableText` for this, given that editable is false? Can we include `ReactMarkdown` directly? I think that would assuage the concern of turning `EditableText` into a footgun?

That makes sense to me!

File: frontend/amundsen_application/static/js/config/config-utils.ts [hunk 1]
```diff
@@ -474,6 +474,13 @@ export function isTableQualityCheckEnabled() {
   return AppConfig.tableQualityChecks.isEnabled;
 }
 
+/**
+ * Returns whether dangerous html should be allowed in programmatic descriptions
+ */
+export function isProgrammaticDescAllowDangerousHtml(): boolean {
+  return AppConfig.programmaticDescriptionAllowDangerousHtml;
+}
+
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/config/config-utils.ts:483]: We need a basic tests for this

File: frontend/amundsen_application/static/js/config/config-utils.ts [hunk 2]
```diff
@@ -474,6 +474,13 @@ export function isTableQualityCheckEnabled() {
   return AppConfig.tableQualityChecks.isEnabled;
 }
 
+/**
+ * Returns whether dangerous html should be allowed in programmatic descriptions
+ */
+export function isProgrammaticDescAllowDangerousHtml(): boolean {
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/config/config-utils.ts:480]: this is a mouthful of name. It is tricky, but I think 'hasHTMLDescription' would work better. Same for the config object field.

File: frontend/amundsen_application/static/js/pages/TableDetailPage/index.tsx [hunk 3]
```diff
@@ -232,7 +233,15 @@ export class TableDetail extends React.Component<
 
     return descriptions.map((d) => (
       <EditableSection key={`prog_desc:${d.source}`} title={d.source} readOnly>
-        <EditableText maxLength={999999} value={d.text} editable={false} />
+        <div className="editable-text">
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/pages/TableDetailPage/index.tsx:236]: I am not sure what are we doing here.
You modified <EditableText> to allow for the 'allowDangerousHtml' prop, why using ReactMarkdown here directly?

**Review by Golodhros — CHANGES_REQUESTED** [review]

**Comment by ozandogrultan** [comment]
@Golodhros I have changed the approach slightly; instead of checking a frontend config variable, now **only programmatic descriptions** always allow dangerous html. This would allow the usage of tags like `iframe` in the programmatic descriptions.

**Comment by Golodhros** [comment]
We talked about this before @dorianj, what do you think about this change?

**Comment by ozandogrultan** [comment]
@dorianj @Golodhros Any updates on this pull request? It is a small change which would help our team (and potentially teams which require similar functionality) a lot and would be great to see it merged.

**Comment by ozandogrultan** [comment]
@dorianj @Golodhros Just bumping this up, our team is still looking forward to seeing this PR getting merged!

**Review by Golodhros — APPROVED** [review]
LGTM

**Pull request merged**